src/steamship/agents/llms/steamship_llm.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from pprint import pformat
+from typing import List, Optional
+
+from steamship import Block, File, MimeTypes, PluginInstance, Steamship, SteamshipError, Tag
+from steamship.agents.schema import Tool
+from steamship.data import GenerationTag, TagKind
+from steamship.plugin.capabilities import (
+    CapabilityPluginRequest,
+    CapabilityPluginResponse,
+    ConversationSupport,
+    FunctionCallingSupport,
+    SystemPromptSupport,
+)
+
+
+class SteamshipLLM:
+    """A class wrapping LLM plugins."""
+
+    def __init__(self, plugin_instance: PluginInstance):
+        # TODO (PR callout): In the spirit of getting folks to pin to specific instances more often, take the route of
+        #  using specific PluginInstances rather than client.use_plugin all the time with the params that we determine?
+        self.client = plugin_instance.client
+        self.plugin_instance = plugin_instance
+
+    @staticmethod
+    def with_gpt4(client: Steamship, temperature: float = 0.4, max_tokens: int = 256):
+        # TODO (PR callout): This is an example of helpers that can initialize plugins that we know are solid / are
+        #  steamship-published.  These don't need to live in this class either, to keep the interface more pure.
+        gpt4 = client.use_plugin(
+            "gpt-4", config={"model": "gpt-4", "temperature": temperature, "max_tokens": max_tokens}
+        )
+        return SteamshipLLM(gpt4)
+
+    def generate(
+        self,
+        messages: List[Block],
+        system_prompt: List[Block] = None,
+        history: Optional[List[Block]] = None,
+        tools: Optional[List[Tool]] = None,
+        assert_capabilities: bool = True,
+        **kwargs,
+    ) -> List[Block]:
+        """
+        Call the LLM plugin's generate method.  Generate requests for plugin capabilities based on input parameters.
+
+        :param messages: Messages to be passed to the LLM to construct the prompt.
+        :param system_prompt: A system prompt to include with the prompt. Providing this requests the
+          SystemPromptSupport capability.
+        :param history: A list of blocks which are previous interactions with the tool. Providing this requests the
+          ConversationSupport capability.
+        :param tools: A list of Tools, representing functions which can be called with this LLM. Providing this requests
+          the FunctionCallingSupport capability, and provides the function definitions for it.
+        :param assert_capabilities: If True (default), raise a SteamshipError if the LLM plugin did not respond with a
+          block that asserts what level capabilities were fulfilled at.
+        :param kwargs: Options that can be passed to the LLM model as other parameters.
+        :return: a List of Blocks that are returned from the plugin.
+        """
+
+        # TODO (PR callout): I'm not certain this class needs to be abstract?  This seems pretty soup-to-nuts.
```

Comment:
Discussion: doesn't seem like SteamshipLLM needs to be an ABC.

Reply:
At a high level, if it's possible to have a single LLM class that really does check the boxes we need, that feels like a great win.

Reply:
There might be an opportunity for naming clarification here if:
- `messages` --> `current_prompt`
- `history` --> `previous_chat_history`

Alternatively, `history` could go away and be replaced with `use_chat_completion: bool`?